lib/fencing/st_client.c

```diff
@@ -195,6 +195,67 @@ stonith_get_namespace(const char *agent, const char *namespace_s)
     return st_namespace_invalid;
 }
 
+gboolean
+watchdog_fencing_enabled_for_node_api(stonith_t *st, const char *node)
+{
+    gboolean rv = FALSE;
+    stonith_t *stonith_api = st?st:stonith_api_new();
+    char *list = NULL;
+
+    if(stonith_api) {
+        if (stonith_api->state == stonith_disconnected) {
+            int rc = stonith_api->cmds->connect(stonith_api, "stonith-api", NULL);
+
+            if (rc != pcmk_ok) {
+                crm_err("Failed connecting to Stonith-API for watchdog-fencing-query.");
+            }
+        }
+
+        if (stonith_api->state != stonith_disconnected) {
+            /* caveat!!!
+             * this might fail when when stonithd is just updating the device-list
+             * probably something we should fix as well for other api-calls */
+            int rc = stonith_api->cmds->list(stonith_api, st_opt_sync_call, STONITH_WATCHDOG_ID, &list, 0);
```

Comment:
Not for this PR, but I wonder if it would be feasible to move the CIB stonith device parsing code to a library, and have the controller keep its own copy of stonith_watchdog_targets to check against, to avoid the overhead of a new stonith connection. There's the possibility of API registration, but that might be irrelevant for this use case.

Reply:
Would be worth a consideration probably.
Atm the way fenced is digesting updates is unfortunately a bit delicate anyway - you remember the thing with rules and stuff based on attributes. So we would have to assure that both sides are picking things up in sync. Explicitly asking fenced is safer here atm but of course the query is overhead.
Yep I do think as well that possibly being able to work with API registered fencing-devices isn't a reason strong enough to justify the queries.